vaadin-checkbox-flow-parent/vaadin-checkbox-flow/src/main/java/com/vaadin/flow/component/checkbox/CheckboxGroup.java

```diff
@@ -99,13 +106,24 @@ public CheckboxGroup() {
                 CheckboxGroup::presentationToModel,
                 CheckboxGroup::modelToPresentation);
         registerValidation();
+
+        addValueChangeListener(e -> validate());
```

Comment:
I'd probably suggest putting this listener under the feature flag check as well to not introduce a breaking change.